editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/BeanConfigUtil.java

```diff
@@ -668,4 +702,37 @@ public void editBeanProperty(Element xmlElement, String name, String value) {
 			xmlElement.setAttribute(GlobalBeanEIP.PROP_VALUE, value);
 		}
 	}
+	
+	public String getClassNameFromReferencedCamelBean(AbstractCamelModelElement selectedEP, String refID) {
+		if (!Strings.isEmpty((String) refID) && selectedEP != null) {
+			Map<String, GlobalDefinitionCamelModelElement> globalDefs = 
+					selectedEP.getCamelFile().getGlobalDefinitions();
+			GlobalDefinitionCamelModelElement referencedBean = globalDefs.get(refID);
+			if (referencedBean instanceof CamelBean) {
+				return ((CamelBean) referencedBean).getClassName();
+			}
+		}
+		return null;
+	}
+	
+	public String[] removeStringFromStringArray(String[] input, String deleteMe) {
+		ArrayList<String> result = new ArrayList<>();
+		for(String item : input)
+			if(!deleteMe.equals(item)) {
+				result.add(item);
+			}
+		return result.toArray(new String[0]);
```

Comment:
maybe return result.toArray(new String[result.size()]); would be better as you did on the other method?